jmx-scraper/src/test/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfigTest.java

```diff
@@ -0,0 +1,407 @@
+  }
+
+  @Test
+  void shouldFailValidation_invalidTargetSystem() {
+    // Given
+    Properties properties = (Properties) validProperties.clone();
+    properties.setProperty(JmxScraperConfig.TARGET_SYSTEM, "hal9000");
+
+    // When and Then
+    assertThatThrownBy(() -> JmxScraperConfig.fromProperties(properties, new Properties()))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageStartingWith("[hal9000] must specify targets from ");
+  }
+
+  @Test
+  void shouldFailValidation_missingOtlpEndpoint() {
+    // Given
+    Properties properties = (Properties) validProperties.clone();
+    properties.remove(JmxScraperConfig.OTLP_ENDPOINT);
+    properties.setProperty(JmxScraperConfig.METRICS_EXPORTER_TYPE, "otlp");
+
+    // When and Then
+    assertThatThrownBy(() -> JmxScraperConfig.fromProperties(properties, new Properties()))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("otel.exporter.otlp.endpoint must be specified for otlp format.");
+  }
+
+  @Test
+  void shouldPassValidation_noMetricsExporterType() throws ConfigurationException {
+    // Given
+    Properties properties = (Properties) validProperties.clone();
+    properties.remove(JmxScraperConfig.OTLP_ENDPOINT);
+    properties.remove(JmxScraperConfig.METRICS_EXPORTER_TYPE);
+
+    // When
+    JmxScraperConfig config = JmxScraperConfig.fromProperties(properties, new Properties());
+
+    // Then
+    assertThat(config).isNotNull();
+  }
+
+  @Test
+  void shouldPassValidation_nonOtlpMetricsExporterType() throws ConfigurationException {
+    // Given
+    Properties properties = (Properties) validProperties.clone();
+    properties.remove(JmxScraperConfig.OTLP_ENDPOINT);
+    properties.setProperty(JmxScraperConfig.METRICS_EXPORTER_TYPE, "logging");
+
+    // When
+    JmxScraperConfig config = JmxScraperConfig.fromProperties(properties, new Properties());
+
+    // Then
+    assertThat(config).isNotNull();
+  }
+
+  @Test
+  void shouldFailValidation_negativeInterval() {
+    // Given
+    Properties properties = (Properties) validProperties.clone();
+    properties.setProperty(JmxScraperConfig.INTERVAL_MILLISECONDS, "-1");
+
+    // When and Then
+    assertThatThrownBy(() -> JmxScraperConfig.fromProperties(properties, new Properties()))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("otel.jmx.interval.milliseconds must be positive.");
+  }
+
+  @Test
+  void shouldFailConfigCreation_invalidInterval() {
+    // Given
+    Properties properties = (Properties) validProperties.clone();
+    properties.setProperty(JmxScraperConfig.INTERVAL_MILLISECONDS, "abc");
+
+    // When and Then
+    assertThatThrownBy(() -> JmxScraperConfig.fromProperties(properties, new Properties()))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("Failed to parse otel.jmx.interval.milliseconds");
+  }
+
+  //  @ClearSystemProperty(key = "otel.metric.export.interval")
```

Comment:
Either remove the commented test code or add a TODO that explains why it's here and the path forward. Thanks.